app/models/comfy/cms/block.rb

```diff
@@ -1,59 +1,70 @@
 class Comfy::Cms::Block < ActiveRecord::Base
   self.table_name = 'comfy_cms_blocks'
 
-  FILE_CLASSES = %w(ActionDispatch::Http::UploadedFile Rack::Test::UploadedFile File).freeze
+  FILE_CLASSES = %w[ActionDispatch::Http::UploadedFile
+                    Rack::Test::UploadedFile File].freeze
 
   attr_accessor :temp_files
 
   # -- Relationships --------------------------------------------------------
   belongs_to :blockable,
-    polymorphic: true
+             polymorphic: true
+
   has_many :files,
-    autosave: true,
-    dependent: :destroy
+           autosave: true,
+           dependent: :destroy
 
   # -- Callbacks ------------------------------------------------------------
   before_save :prepare_files
 
   # -- Validations ----------------------------------------------------------
-  validates :identifier, :presence => true
+  validates :identifier, presence: true
 
   # -- Instance Methods -----------------------------------------------------
   # Tag object that is using this block
   def tag
-    @tag ||= blockable.tags(:reload).detect{|t| t.is_cms_block? && t.identifier == identifier}
+    @tag ||= blockable.tags(:reload).detect do |t|
+      t.is_cms_block? && t.identifier == identifier
+    end
   end
 
   # Intercepting assigns as we can't cram files into content directly anymore
   def content=(value)
     self.temp_files = [value].flatten.select do |f|
       FILE_CLASSES.member?(f.class.name)
     end
-
     # correctly triggering dirty
-    value = nil               if self.temp_files.present?
-    write_attribute(:content, value)
-    self.content_will_change! if self.temp_files.present?
+    if temp_files.present?
+      write_attribute(:content, nil)
+      content_will_change!
+    else
+      write_attribute(:content, value)
+    end
   end
 
   protected
 
   # If we're passing actual files into content attribute, let's build them.
   def prepare_files
-    return if self.temp_files.blank?
-
-    # only accepting one file if it's PageFile. PageFiles will take all
-    single_file = self.tag.instance_of?(ComfortableMexicanSofa::Tag::PageFile)
-    self.temp_files = [self.temp_files.first].compact if single_file
+    return if temp_files.blank?
 
-    self.temp_files.each do |file|
-      self.files.collect{ |f| f.mark_for_destruction } if single_file
+    restrict_to_single_file! if pagefile?
 
-      self.files.build(
-        site: self.blockable.site,
-        dimensions: self.tag.try(:dimensions),
+    temp_files.each do |file|
+      files.build(
+        site: blockable.site,
+        dimensions: tag.try(:dimensions),
         file: file
       )
     end
   end
+
+  def pagefile?
+    tag.instance_of?(ComfortableMexicanSofa::Tag::PageFile)
+  end
+
+  def restrict_to_single_file!
+    self.temp_files = [temp_files.first].compact
+    files.map(&:mark_for_destruction)
```

Comment:
```ruby
file.each(&:mark_for_destruction)
```

should work fine if you don't need to build a new array with `map`, but just return the original.

Again... huuuge improvement 💯

Reply:
oh true! Good catch! :+1: